Disable env access and file load operations in yqlib

jandubois opened the pull request:
Before:

```console
❯ l ls --yq 'env(HOME)'
/Users/jan

❯ l ls --yq 'load("/usr/local/share/lima/templates/ubuntu.yaml")'
{
    "minimumLimaVersion": "2.0.0",
    "base": [
        "template:_images/ubuntu-25.10",
        "template:_default/mounts"
    ]
}
```

After:

```console
❯ l ls --yq 'env(HOME)'
FATA[0000] env operations have been disabled

❯ l ls --yq 'load("/usr/local/share/lima/templates/ubuntu.yaml")'
FATA[0000] file operations have been disabled
```

Addresses objections to using YQ expressions for `limactl list --filter` in #4187.

jandubois commented on hack/bats/tests/list.bats:
```diff
@@ -264,3 +264,13 @@ local_setup() {
     assert_output "foo"
 }
+
+@test '--yq cannot access environment variables' {
```
I think it is best to restrict it everywhere, except for `limactl yq` itself. 

What is the use case for `limactl list --yq` or `limactl info --yq` to use env variables or load from files?

jandubois replied:
> Env can be useful for parametizing `limactl (create|start|edit) --set`.

This can easily be done by variable interpolation in the shell, or whatever program is invoking `limactl`.

I would still prefer to have a single YQ dialect that applies to all `limactl` options instead of documenting each one individually which operations are available.

I don't understand the desire to add conceptual complexity without a compelling use-case. The only reason to make this available is because the operators already exist. If they didn't, would we open an issue against YQ asking for them to be added?

AkihiroSuda replied:
> I would still prefer to have a single YQ dialect that applies to all limactl options instead of documenting each one individually which operations are available.

When `limactl` is invoked with `yq` as `argv[0]`, it has to emulate the upstream `yq` behavior.

In the current implementation, the yq-emulating mode might not be using `yqutil.EvaluateExpressionWithEncoder`, but the implementation can change in future and it is hard to track that what "dialect" is applied to the global yqlib configuration.

We do not need to introduce any "dialect" if we avoid using yq in implementing `--filter`.
Instead of yq, containerd's filter package could be used: https://github.com/lima-vm/lima/pull/4187#discussion_r2671848242

AkihiroSuda replied:
Let's merge this PR after:
- Add test for the yq multi-call binary
- Add a document to explain our "dialect" of yq language

jandubois commented:
> Let's merge this PR after:
> 
> * Add test for the yq multi-call binary

I've rebased the PR on latest `master` and added tests for the multi-call functionality.

> * Add a document to explain our "dialect" of yq language

I'm struggling to find a good place in the documentation for this. Like a place users would actually find.

I feel like we also need more documentation for the `create --set` option, including examples. This could go in the `limactl create --help` output, and could then mention the limitation on env and file operators in our YQ dialect. Other options could then point to it.

This might still be a bit unwieldy, so maybe we should start having "concept guides" the way `git` does it:

```console
❯ git help -g
The Git concept guides are:
   core-tutorial    A Git core tutorial for developers
   credentials      Providing usernames and passwords to Git
   cvs-migration    Git for CVS users
   diffcore         Tweaking diff output
   everyday         A useful minimum set of commands for Everyday Git
   faq              Frequently asked questions about using Git
   glossary         A Git Glossary
   namespaces       Git namespaces
   remote-helpers   Helper programs to interact with remote repositories
   submodules       Mounting one repository inside another
   tutorial         A tutorial introduction to Git
   tutorial-2       A tutorial introduction to Git: part two
   workflows        An overview of recommended workflows with Git

'git help -a' and 'git help -g' list available subcommands and some
concept guides. See 'git help <command>' or 'git help <concept>'
to read about a specific subcommand or concept.
See 'git help git' for an overview of the system.
```

There is also `kubectl options` as another precedent; not sure if it has other meta commands.

Anyways, this is quickly becoming out-of-scope for this PR and should be in a separate docs PR.

AkihiroSuda commented:
Is this PR still a draft?

> concept guides

This can be probably implemented using https://pkg.go.dev/github.com/spf13/cobra#Command.IsAdditionalHelpTopicCommand

jandubois commented:
> Is this PR still a draft?

I turned it into a draft because we said yesterday it needs documentation before it can be merged, and the documentation is still outstanding (see my comments above).

The code is complete, if you are willing to create an issue for the docs and merge the code as-is.
